include/swift/AST/SwiftNameTranslation.h

```diff
@@ -99,11 +102,17 @@ struct DeclRepresentation {
 };
 
 /// Returns the C++ representation info for the given declaration.
-DeclRepresentation getDeclRepresentation(const ValueDecl *VD);
+DeclRepresentation getDeclRepresentation(
+    const ValueDecl *VD,
+    std::optional<std::function<bool(const NominalTypeDecl *)>> isZeroSized =
+        std::nullopt);
```

Comment:
Is it reasonable to invoke this without passing a zero-size check in some cases? Wondering if the default value of the parameter is needed here.

Reply:
I don't feel strongly, but could we remove the default value on the declaration, and pass `nullopt` on that call site, since that seems to be an exception that we wouldn't want to be repeated elsewhere?

Reply:
Ah, good point! Will do.